File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let prv = [...arr];
    let result = 0;
    while (true) {
        let nxt = []
        for (let i=0; i<prv.length; i++) {
            if (prv[i] >= 50 && prv[i] % 2 === 0) {
                nxt.push(Math.floor(prv[i] / 2));
            } else if (prv[i] < 50 && prv[i] % 2 !== 0) {
                nxt.push(prv[i] * 2 + 1);
            } else {
                nxt.push(prv[i]);
            }
        }
        if (JSON.stringify(prv) === JSON.stringify(nxt)) { break }
        prv = [...nxt];
        result += 1;
    }
    return result;
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    prv = arr[:]
    x = 0
    while True:
        nxt = []
        for i in prv:
            if i >= 50 and i % 2 == 0:
                nxt.append(i // 2)
            elif i < 50 and i % 2 != 0:
                nxt.append(i * 2 + 1)
            else:
                nxt.append(i)
        if prv == nxt:
            break
        prv = nxt[:]
        x += 1
    return x